# SVHack_group3

Impact of bacteria diversity on alignment accuracy for antibiotic resistant organisms
![Blank diagram](https://github.com/collaborativebioinformatics/SVHack_metagenomics/assets/72709799/a7a2a040-aa64-48a1-9650-482cbbc3e647)



![Flow_chart_Day1](https://github.com/collaborativebioinformatics/SVHack_metagenomics/assets/72709799/f22e93af-ff21-4df4-a173-1ed5422bc7b9)


Task A : Graph genome Construction 

Task A1: Download genome assemblies full-length genome, diverse strains isolated from different environments (find assembly accession and make a metadata table).  Tania, Tony
Task A2: Constructing graph genome ( find graph construction tools) e.g. minigraph. (using 5/20/100/500 assemblies). Jason/ Tania, Shaheerah, Pedro
Task A3: Choose an aligner specific for graph genomes, and perform alignments. Alignments will be performed using the reads obtained in Task B.  Astrid, Lorianne
Task A4: Choose a method for visualization and generating stats and use it.
Task A5: Calculate statistics from alignments (time/#aligned reads/quality distribution)
Task A6: Generate graph genome annotation (Optional).

Task B : Standard genome pipeline

Task B1: Download reference genome. Muhammad
Task B2: Download the illumina DNA sequencing reads (different biosamples than those used for the graph genome). Muhammad
Task B3: Align reads to the reference genome from B1 using BWA MEM, Shaheereh, Pedro
Task B4: Calculate statistics from alignments (time/#aligned reads/quality distribution).

